src/nodes.rs

```diff
@@ -49,7 +49,7 @@ impl Div<usize> for NodeCount {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq, From, Encode, Decode)]
-pub(crate) struct NodeMap<T>(Vec<T>);
+pub struct NodeMap<T>(Vec<T>);
```

Comment:
I didn't mean to make the `NodeMap` public, but deriving `Decode` causes a leak of that type. I added a custom implementation od `Encode + Decode` to avoid leaking the type.

Reply:
Also, on a side note: the type `T` in `NodeMap<T>` is currently always `Option<_>`. Probably we should refactor `NodeMap<T>` so that it has an interface more similar to say `HashMap<T>` (an element for an index can be absent), and also `BoolNodeMap` should be renames to `NodeSet`